docs(replaceFieldWriting): improve the documentation for the replaceFieldWriting (#334)
ForestAdmin/agent-nodejs: docs/agent-customization/fields/write.md

### Substitution

```javascript
collection.replaceFieldWriting('fullName', (value) => {
  const [firstName, lastName] = value.split(' ');

  return { firstName, lastName };
});
```

> #### Trigger what you want

Reviewer: ```suggestion
#### Trigger your own code
```

> You can also trigger any code you want without update the field by returning nothing.

Reviewer: ```suggestion
You can also trigger any code you want without updating any field by not returning anything.
```

```javascript
collection.replaceFieldWriting('fullName', async (fullName) => {
  await EmailSender.send(`${fullName} wants to be update`);
});
```

> #### The relationships

Reviewer: ```suggestion
#### Handling relationships
```

{% hint style="info" %}

> These tricks only work on the `Many to One` and `One to One` relationships.

Reviewer: ```suggestion
Handling relationships inside a `replaceFieldWriting` will only work for `Many to One` and `One to One` relationships.
```

{% endhint %}

In this simple example, we have two collections which are linked together:

- `Users` has a `job` and a `portfolioId` as foreignKey 
- `Portfolios` has a `title`

When the user updates his `job` field we want also update the `title` of the portfolio by the `job` name.

```javascript
collection.replaceFieldWriting('job', (job, { action }) => {

>   // it will update the title of the right portfolio.

Reviewer: ```suggestion
```

IMO, you already mentionned what the code is doing above, so no need for this comment

  return { job, portfolio: { title: job } };
});
```

{% hint style="info" %}
If the relationships does not exist it will create it with the given fields values.
{% endhint %}

You can also provide another `portfolioId` to update the relationships and its fields:

```javascript
collection.replaceFieldWriting('job', (job, { action }) => {

>   // it will update the portfolio's title with the id 8 and updates also the foreignkey.

Reviewer: ```suggestion
```

Same as above, IMO this is not needed

  return { job, portfolioId: 8, portfolio: { title: job } };
});
```

> Of course you can chained the relationships, for example, if a portfolio has a `one to one` relationships
> with the `formats` collection, you can update it by writing the right path.

Reviewer: ```suggestion
Of course you can chain the relationships. For example, if a portfolio has a `one to one` relationship
with the `formats` collection, you can update it by writing the right path.
```

```javascript
collection.replaceFieldWriting('job', (job, { action }) => {
  return { job, portfolioId: 8, portfolio: { title: job, format: { name: 'pdf' } } };
});
```

#### Get the action type from the context

> When you want to trigger a different code when the field is `updated` or `created`
> you can use the `action` field from the `context`.

Reviewer: ```suggestion
If you want to trigger a different code when the field is `updated` or `created`
you can use the `action` field from the `context`.
```

collection.replaceFieldWriting('fullName', null);
```

> ## Substitution

Reviewer: ```suggestion
## Substitution

By default, he `replaceFieldWriting` function allow you to substitute any given fields by returning a patch of the current record.

In the following example, editing or creating a `fullName` will update both `firstName` and `lastName` fields of the record.

```

```javascript
collection.replaceFieldWriting('fullName', (value) => {
  const [firstName, lastName] = value.split(' ');

  return { firstName, lastName };
});
```

> ### Trigger your own code

Reviewer: This should either be a whole `##` title, or we wrapped in a subsection with a `##` title?
